napari/plugins.py

```diff
@@ -0,0 +1,127 @@
+import importlib
+import pkgutil
+import re
+
+import pkg_resources
+import requests
+
+"""
+This proposal for plugin discovery follows two of the recommendations here:
+https://packaging.python.org/guides/creating-and-discovering-plugins/
```

Comment:
Conforming to the recommendations from https://packaging.python.org/guides/creating-and-discovering-plugins/ certainly makes sense to me